redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Node.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood
+
+/**
+ * An interface that has the common behavior as a node contained in a flexbox.
+ */
+public interface Node {
```

Comment:
Can we make this into a class? Perhaps with a user-defined tag value?

Reply:
I assume the only reason this can’t be an interface today is so we can have a polymorphic `measure` function. But perhaps that can be an external adapter thing?

One thing that I’d like to try is separating the value-object bits from the service bits. If I apply my value/service/glue ideal, this is a platypus. I wonder if we could split out responsibilities differently.

Reply:
Yep, the API currently doesn't have a way to mark nodes as dirty. It assumes a node's properties don't change once it's added to the layout. The plan is to add this later after adding some more regression tests.